### PR TITLE
Add task detail view with navigation, messaging, and log history

### LINKED ISSUE #777: Add task detail view with navigation and logs
## Summary
Build the task detail view UI with navigation from the task list, real-time logs, and adaptive polling based on task state.

## Goals
- Detail view showing task status, logs, and actions
- Navigation between task list and detail view
- Real-time logs via push messages
- Adaptive polling intervals (faster for active tasks, slower for idle/complete)
- Message input for interacting with tasks

## Resources
- [Figma designs](https://www.figma.com/design/l3PZmjWg7I9Llzd5VepNCV/VS-Code-extension-for-Tasks)
- [Notion spec](https://www.notion.so/coderhq/Enable-Tasks-in-the-VS-Code-Extension-2bdd579be59280ddba04cd5c3cdb354c)

### PR DESCRIPTION
**Pull request by EhabY**
## Summary

Clicking a task in the list now navigates to a detail page with header, log history, and message input.

### Components
- **TaskDetailView**: composes the header, chat history, error banner, and message input
- **TaskDetailHeader**: back button, status dot, task label with loading text (e.g. "Pausing..."), action menu
- **AgentChatHistory**: log entries grouped by role (user/agent) with auto-scroll via `useFollowScroll`
- **TaskMessageInput**: placeholder changes per task status, shows a pause button when the agent is working, Ctrl+Enter to send
- **ErrorBanner**: error message with "View logs" link, shown when task status is error

### Hooks
- **useSelectedTask**: tracks selected task, validates against the task list, fetches details with adaptive polling (faster when active, slower when idle)
- **useFollowScroll**: keeps chat scrolled to bottom as logs grow, backs off when the user scrolls up

### Other changes
- `sendTaskMessage` promoted from command to request (now returns a response)
- `canSendMessage` added to `TaskPermissions`
- `taskDetails()` test factory and Proxy-based `useTasksApi` mock for cleaner tests
- Tests for all new components and utilities

Closes #777

### CODE REVIEW

**Comment by EhabY** [comment]
Created this as a followup https://github.com/coder/vscode-coder/issues/793

File: packages/shared/src/tasks/api.ts [hunk 1]
```diff
@@ -47,13 +47,13 @@ const deleteTask = defineRequest<TaskActionParams, void>("deleteTask");
 const pauseTask = defineRequest<TaskActionParams, void>("pauseTask");
 const resumeTask = defineRequest<TaskActionParams, void>("resumeTask");
 const downloadLogs = defineRequest<{ taskId: string }, void>("downloadLogs");
+const sendTaskMessage = defineRequest<
+	{ taskId: string; message: string },
```
Comment (mtojek) [inline packages/shared/src/tasks/api.ts]: q about consistency: pauseTask and resumeTask take `TaskActionParams`, should we use a similar pattern here as well?

File: packages/shared/src/tasks/utils.ts [hunk 2]
```diff
@@ -26,10 +26,14 @@ const RESUMABLE_STATUSES: readonly TaskStatus[] = [
 export function getTaskPermissions(task: Task): TaskPermissions {
 	const hasWorkspace = task.workspace_id !== null;
 	const status = task.status;
+	const canSendMessage =
+		task.status === "paused" ||
+		(task.status === "active" && task.current_state?.state !== "working");
```
Comment (mtojek) [inline packages/shared/src/tasks/utils.ts:36]: possibly missing test: active status and current state is null

File: packages/tasks/src/components/AgentChatHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+							key={log.id}
+							log={log}
+							isGroupStart={index === 0 || log.type !== logs[index - 1].type}
+						/>
+					))
+				)}
+				{isThinking && (
+					<div className="log-entry log-entry-thinking">Thinking...</div>
+				)}
+				<div ref={bottomRef} />
```
Comment (mtojek) [inline packages/tasks/src/components/AgentChatHistory.tsx:65]: missing "key"?

File: packages/tasks/src/components/AgentChatHistory.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			</VscodeScrollable>
+		</div>
+	);
+}
+
+function getEmptyMessage(logsStatus: LogsStatus): string {
+	switch (logsStatus) {
```
Comment (mtojek) [inline packages/tasks/src/components/AgentChatHistory.tsx:72]: hmm.. shouldn't we handle "ok" for clarity?

File: packages/tasks/src/components/ErrorBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	const message = task.current_state?.message || "Task failed";
+
+	return (
+		<div className="error-banner">
+			<VscodeIcon name="warning" />
+			<span>{message}</span>
+			<button
+				type="button"
+				className="text-link"
+				onClick={() => api.viewLogs(task.id)}
```
Comment (mtojek) [inline packages/tasks/src/components/ErrorBanner.tsx:22]: is it safe if api.viewLogs throws an exception?

File: packages/tasks/src/components/PromptInput.tsx [hunk 6]
```diff
@@ -47,10 +51,10 @@ export function PromptInput({
...
 					<VscodeIcon
 						actionIcon
-						name="send"
-						label="Send"
-						onClick={() => canSubmit && onSubmit()}
-						className={canSubmit ? "" : "disabled"}
+						name={actionIcon}
+						label={actionLabel}
+						onClick={() => !actionDisabled && onSubmit()}
+						className={actionDisabled ? "disabled" : ""}
```
Comment (mtojek) [inline packages/tasks/src/components/PromptInput.tsx]: nit: it is a good habit to keep positive wording, so "actionEnabled". Phrases like `!actionDisabled` are cognitively harder to digest :)

File: packages/tasks/src/components/TaskDetailView.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+interface TaskDetailViewProps {
+	details: TaskDetails;
+	onBack: () => void;
+}
+
+export function TaskDetailView({ details, onBack }: TaskDetailViewProps) {
+	const { task, logs, logsStatus } = details;
+
+	const isThinking =
```
Comment (mtojek) [inline packages/tasks/src/components/TaskDetailView.tsx]: nit: should we extract it to `utils`?

File: packages/tasks/src/components/TaskMessageInput.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+		mutationFn: (msg: string) => api.sendTaskMessage(task.id, msg),
+		onSuccess: () => setMessage(""),
+	});
+
+	return (
+		<PromptInput
+			placeholder={placeholder}
+			value={message}
+			onChange={setMessage}
+			onSubmit={showPauseButton ? pauseTask : () => sendMessage(message)}
```
Comment (mtojek) [inline packages/tasks/src/components/TaskMessageInput.tsx:71]: q: is it possible to quickly click the button twice or will UI block this behavior?

File: packages/tasks/src/components/TaskMessageInput.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+	const canSubmitMessage = canSendMessage && message.trim().length > 0;
+
+	const { mutate: pauseTask, isPending: isPausing } = useMutation({
+		mutationFn: () =>
+			api.pauseTask({ taskId: task.id, taskName: getTaskLabel(task) }),
+	});
+
+	const { mutate: sendMessage, isPending: isSending } = useMutation({
+		mutationFn: (msg: string) => api.sendTaskMessage(task.id, msg),
+		onSuccess: () => setMessage(""),
```
Comment (mtojek) [inline packages/tasks/src/components/TaskMessageInput.tsx:62]: hmm.. isn't onError missing?

File: packages/tasks/src/hooks/useFollowScroll.ts [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+			}
+		});
+		ro.observe(container);
+
+		return () => {
+			container.removeEventListener("vsc-scrollable-scroll", onScroll);
+			mo.disconnect();
+			ro.disconnect();
+		};
+	}, []);
```
Comment (mtojek) [inline packages/tasks/src/hooks/useFollowScroll.ts:65]: no deps here?

File: packages/tasks/src/App.tsx [hunk 11]
```diff
@@ -95,14 +100,17 @@ export default function App() {
...
 			>
-				<VscodeScrollable ref={historyScrollRef}>
-					<TaskList
-						tasks={tasks}
-						onSelectTask={(_taskId: string) => {
-							// Task detail view will be added in next PR
-						}}
-					/>
-				</VscodeScrollable>
+				<div ref={historyScrollRef} className="collapsible-content">
```
Comment (mtojek) [inline packages/tasks/src/App.tsx:117]: nit: alternative form, perhaps easier to read

```tsx
function renderHistory() {
    if (selectedTask) return <TaskDetailView .../>;
    if (isLoadingDetails) return <div className="loading-container">...</div>;
    return <TaskList .../>;
}
```

File: src/webviews/tasks/tasksPanel.ts [hunk 12]
```diff
@@ -355,10 +356,44 @@ export class TasksPanel
...
+				task.template_version_id,
+			);
+		}
+
+		try {
+			await this.client.sendTaskInput("me", taskId, message);
+		} catch (err) {
+			if (
+				isAxiosError(err) &&
+				(err.response?.status === 409 || err.response?.status === 400)
```
Comment (mtojek) [inline src/webviews/tasks/tasksPanel.ts:385]: shouldn't we log somewhere what was the real http code and message?

File: package.json [hunk 13]
```diff
@@ -219,7 +219,7 @@
 					"id": "coder.tasksPanel",
 					"name": "Coder Tasks",
 					"icon": "media/tasks-logo.svg",
-					"when": "coder.authenticated && coder.tasksEnabled"
+					"when": "coder.tasksEnabled"
```
Comment (mtojek) [inline package.json:222]: curious: why?

**Review by mtojek — COMMENTED** [review]
Leaving a bunch of comments, hopefully nothing big. 👍 👍 for adding tests!

File: packages/shared/src/tasks/api.ts (hunk 1 above)
Comment (EhabY) [reply packages/shared/src/tasks/api.ts]: We could and done!

File: packages/shared/src/tasks/utils.ts (hunk 2 above)
Comment (EhabY) [reply packages/shared/src/tasks/utils.ts:36]: Added test case

File: packages/tasks/src/components/AgentChatHistory.tsx (hunk 3 above)
Comment (EhabY) [reply packages/tasks/src/components/AgentChatHistory.tsx:65]: This is outside the `logs.map`, also the eslint rules would catch it like this:

> Missing "key" prop for element in iterator

File: packages/tasks/src/components/AgentChatHistory.tsx (hunk 4 above)
Comment (EhabY) [reply packages/tasks/src/components/AgentChatHistory.tsx:72]: Done!

File: packages/tasks/src/components/ErrorBanner.tsx (hunk 5 above)
Comment (EhabY) [reply packages/tasks/src/components/ErrorBanner.tsx:22]: This here just calls:

```ts
postMessage({
		method: "viewLogs",
		params: task.id,
});
```

Which is VS Code API

All requests go through `useMutation` which has an `onError` (since those are async). `command` are synchronous and called in line, I'll add some error handling around commands and make sure everything goes through the `logger` actually!

File: packages/tasks/src/components/PromptInput.tsx (hunk 6 above)
Comment (EhabY) [reply packages/tasks/src/components/PromptInput.tsx]: Ah, good point, changed :)!

File: packages/tasks/src/components/TaskDetailView.tsx (hunk 7 above)
Comment (EhabY) [reply packages/tasks/src/components/TaskDetailView.tsx]: True actually, done!

File: packages/tasks/src/components/TaskMessageInput.tsx (hunk 9 above)
Comment (EhabY) [reply packages/tasks/src/components/TaskMessageInput.tsx:62]: It was yes, I went through these and fixed them 🙏

File: packages/tasks/src/components/TaskMessageInput.tsx (hunk 8 above)
Comment (EhabY) [reply packages/tasks/src/components/TaskMessageInput.tsx:71]: `canSendMessage`/`sendMessage` are both outputs from `useMutation`, `canSendMessage` disabled the button so it cannot be clicked.

It's blocked in the UI and in the event handler as well

File: packages/tasks/src/hooks/useFollowScroll.ts (hunk 10 above)
Comment (EhabY) [reply packages/tasks/src/hooks/useFollowScroll.ts:65]: There are no changing dependencies here (the `ref` objects are stable) + We only want to run this once on mount which is what the `[]` communicates

File: packages/tasks/src/App.tsx (hunk 11 above)
Comment (EhabY) [reply packages/tasks/src/App.tsx:117]: Oh I like this, done!

File: src/webviews/tasks/tasksPanel.ts (hunk 12 above)
Comment (EhabY) [reply src/webviews/tasks/tasksPanel.ts:385]: We already log in the interceptor the full error, but I can enhance this as follows:

```
throw new Error(
					`Task is not ready to receive messages (${errToStr(err)})`,
				);
```

File: package.json (hunk 13 above)
Comment (EhabY) [reply package.json:222]: We already had a "Welcome screen" that shows when we are not authenticated so this was wrong from the beginning!

File: packages/tasks/src/components/AgentChatHistory.tsx (hunk 3 above)
Comment (mtojek) [reply packages/tasks/src/components/AgentChatHistory.tsx:65]: TIL, thanks!

File: packages/tasks/src/components/AgentChatHistory.tsx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+		case "not_available":
+			return "Logs not available in current task state";
+		case "error":
+			return "Failed to load logs";
+		case "ok":
```
Comment (mtojek) [inline packages/tasks/src/components/AgentChatHistory.tsx:77]: Thanks for adding this! Isn't it a problem that we don't have the `default` case? In such cases, I used to add a logger warning about unexpected state.

**Review by mtojek — APPROVED** [review]
Ship it!

File: packages/tasks/src/components/AgentChatHistory.tsx (hunk 14 above)
Comment (EhabY) [reply packages/tasks/src/components/AgentChatHistory.tsx:77]: It's okay as long as the type is exhaustive - So if we add another type like "unexpected", this part would not compile even so we have to update it
